core/model/generation.go

```diff
@@ -14,6 +18,18 @@ const (
 	generationKeySuffix = "#next"
 )
 
+// ValidateBranchName returns an error if the input name is not suitable for
+// identifying a new in-flight branch.
+func ValidateBranchName(name string) error {
+	if name == "" {
+		return errors.NotValidf("empty branch name")
+	}
+	if name == GenerationMaster {
+		return errors.NotValidf("branch name %q", GenerationMaster)
```

Comment:
You should mention this in the description of creating a branch. Something along the lines as, there are reserved branch names, rather than just erroring later on.

Reply:
Good point; done.